designs/administrator/future-openshift/bulk-actions/index.md

```diff
@@ -0,0 +1,139 @@
+---
+parent: Administrator
+version: 4.x
+---
+
+# Bulk actions
+
+Bulk actions will be added to a number of list views in the OpenShift console. This feature will allow users to perform an action on multiple resources at once. The list of actions include adding labels, adding annotations and deleting resources.
+
+## Bulk actions in toolbar
+The bulk selector will be the first component in the toolbar on pages where bulk actions are available.
+![toolbar](img/toolbar.png)
+
+Users can click the checkbox in the bulk selector to select all items in the list. If all items are selected, the bulk selector checkbox will be filled with a check and the bulk selector will read ‘[#] selected’, informing users that all known items in the list are selected. Clicking the bulk selector checkbox again will deselect all items.
+![bulk-selector-states](img/bulk-selector-states.png)
+
+Users can also choose to select a group of items by clicking the associated checkboxes next to them. The bulk selector checkbox will be filled in with a ‘-‘ to indicate that some items are selected, and the bulk selector will display the number of items selected (e.g. ‘2 selected’). Clicking the bulk selector checkbox will deselect all items.
+![add-labels](img/add-labels.png)
+
+## Behavior
+If no items are selected, the bulk actions dropdown will be enabled, shown as a basic dropdown, but the actions within will be disabled. A tooltip will explain to users that they must select resources in order to take actions.
+![none-checked](img/none-checked.png)
+
+When items are selected, the bulk actions basic dropdown will turn into a primary dropdown, and actions within it will be enabled. 
+![add-labels](img/add-labels.png)
+
+### Item limits
+Depending on the use case, there may be scenarios where users are limited to selecting a certain number of items, assuming there is a large total in a dataset. As a general rule of thumb, the max number of items that can be selected at once will be set at 200. 
+![max-number-200](img/max-number-200.png)
+
+Once the user hits that limit, they will not be able to select more items and all other checkboxes will be disabled. 
+![max-number-error](img/max-number-error.png)
+
+### Permission limits
+Depending on the use case, there may be scenarios where users are limited to a certain set of actions to perform on items. If the user does not have permission to perform a specific action on **any** of the selected items, the action will still be enabled in the dropdown. Once they select the action, the modal will contain a failure alert, explaining that the action cannot be taken.
+![no-permissions](img/no-permissions.png)
```

Comment:
This is sort of a conventions-related thing but I have seen other designs that have modals containing just an inline alert as well, but something about it to me seems like its not really "inline" at that point anymore? Could the whole modal not just be the alert contents? The modal title "This action cannot be taken" with the icon to the left of it and the message as its contents? Just something I had thought about recently.

Reply:
Yes definitely something we could do - like we do with other error/warning states within modals. I can change the design and reflect that in this! @itsptk